### PR TITLE
Awkward Arrays and Wranglers

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
### Changes in API
`Word2VecTransformer` now takes a list of documents instead of a list of sentences, meaning one level of nesting has been added.
```python
# old signature:
def transform(self, sentences: Iterable[Iterable[str]]):
    pass

# new signature:
def transform(self, documents: Iterable[Iterable[Iterable[str]]]):
    pass
```
This is useful, as now you can pipe `SpacyPreprocessor` and `Word2VecTransformer` directly into each other.
```python
from sklearn.pipeline import make_pipeline

pipeline = make_pipeline(SpacyPreprocessor(nlp), Word2VecTransformer())
``` 
`Word2VecTransformer` now outputs a ragged *Awkward Array* instead of `List[List[np.ndarray]]`.
This is great as now you can directly do arithmetic and all sorts of operations on these arrays.
What's also great about both of these changes is that the internal structure of the data remains paramount, therefore you can uniquely identify documents in the transformer's output.

### Wranglers

Wrangler components are now part of the codebase, which you can use for manipulating (potentially ragged) arrays in a machine learning pipeline.
Three wranglers have been added: `Flattener`, which flattens arrays along a given axis, `Pooler` which can reduce the dimensionality of arrays with an aggregation and `Padder` which can pad ragged arrays along a given axis.
These are rather useful, because you can easily manipulate the output of the `Word2VecTransformer` to be in a particular format that you might want to use in following pipeline components, let's say regression or classification models.
Here's a quick example of how you would build a pipeline, that takes the mean of all vectors in a documents, thereby giving a single embedding for each of them.

```python
pipeline = make_pipeline(
    SpacyPreprocessor(nlp),
    Word2VecTransformer(),
    # Pooling sentences together (aka. taking mean of all embeddings)
    Pooler(axis=2),
    # Pooling all sentences in a document
    # axis=1 is the default.
    Pooler()
)
```

### CODE REVIEW

**Comment by jankounchained** [comment]
super thumbs up for the wrangling methods and Awkward Arrays.

File: skword2vec/preprocessing/__init__.py
Comment (x-tabdeveloping) [inline skword2vec/preprocessing/__init__.py:1]: I'm not exactly sure about these two. You could imagine that it might be advantageous for someone using this not to have spaCy/gensim installed, and just make them an optional dependency. That way they only need to have either of them if they import Word2VecTransformer or SpacyPreprocessor.
